Use arrow functions for anonymous functions where possible (part #8) (#19191)
mdn/content: files/en-us/web/api/web_workers_api/using_web_workers/index.md

```js
function QueryableWorker(url, defaultListener, onError) {
  const instance = this;
  const worker = new Worker(url);
  const listeners = {};

>   this.defaultListener = defaultListener || () => {};

Reviewer: ```suggestion
  this.defaultListener = defaultListener ?? (() => {});
```

  if (onError) { worker.onerror = onError; }

  this.postMessage = (message) => {
    worker.postMessage(message);
  }

  this.terminate = () => {
    worker.terminate();
  }
}
```

Then we add the methods of adding/removing listeners:

```js
this.addListeners = (name, listener) => {
  listeners[name] = listener;
}

this.removeListeners = (name) => {
  delete listeners[name];
}
```

Here we let the worker handle two simple operations for illustration: getting the difference of two numbers and making an alert after three seconds. In order to achieve that we first implement a `sendQuery` method which queries if the worker actually has the corresponding methods to do what we want.

```js

> // This functions takes at least one argument, the method name we want to query.
> // Then we can pass in the arguments that the method needs.
> this.sendQuery = () => {
>   if (arguments.length < 1) {
>     throw new TypeError('QueryableWorker.sendQuery takes at least one argument');
>   }
>   worker.postMessage({
>     'queryMethod': arguments[0],
>     'queryMethodArguments': Array.prototype.slice.call(arguments, 1)
>   });
> }

Reviewer: ```suggestion
// This functions takes at least one argument, the method name we want to query.
// Then we can pass in the arguments that the method needs.
this.sendQuery = (method, ...args) => {
  if (!method) {
    throw new TypeError('QueryableWorker.sendQuery takes at least one argument');
  }
  worker.postMessage({
    queryMethod: method,
    queryMethodArguments: args,
  });
}
```

```html
<!doctype html>
<html lang="en-US">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width">
    <title>MDN Example - Queryable worker</title>
    <script type="text/javascript">
      // QueryableWorker instances methods:
      //   * sendQuery(queryable function name, argument to pass 1, argument to pass 2, etc. etc): calls a Worker's queryable function
      //   * postMessage(string or JSON Data): see Worker.prototype.postMessage()
      //   * terminate(): terminates the Worker
      //   * addListener(name, function): adds a listener
      //   * removeListener(name): removes a listener
      // QueryableWorker instances properties:
      //   * defaultListener: the default listener executed only when the Worker calls the postMessage() function directly
      function QueryableWorker(url, defaultListener, onError) {
        const instance = this,
        const worker = new Worker(url),
        const listeners = {};

>         this.defaultListener = defaultListener || () => {};

Reviewer: ```suggestion
        this.defaultListener = defaultListener ?? (() => {});
```

        if (onError) { worker.onerror = onError; }

        this.postMessage = (message) => {
          worker.postMessage(message);
        }

        this.terminate = () => {
          worker.terminate();
        }

        this.addListener = (name, listener) => {
          listeners[name] = listener;
        }

        this.removeListener = (name) => {
          delete listeners[name];
        }

        // This functions takes at least one argument, the method name we want to query.
        // Then we can pass in the arguments that the method needs.

>         this.sendQuery = () => {

Reviewer: Same as above; need to be refactored into `(method, ...args)`

      // This script WON'T be parsed by JS engines because its MIME type is text/js-worker.
      const myVar = 'Hello World!';
      // Rest of your worker code goes here.
    </script>
    <script>
      // This script WILL be parsed by JS engines because its MIME type is text/javascript.
      function pageLog(sMsg) {
        // Use a fragment: browser will only render/reflow once.
        const frag = document.createDocumentFragment();
        frag.appendChild(document.createTextNode(sMsg));
        frag.appendChild(document.createElement('br'));
        document.querySelector('#logDisplay').appendChild(frag);
      }
    </script>
    <script type="text/js-worker">
      // This script WON'T be parsed by JS engines because its MIME type is text/js-worker.
      onmessage = (event) => {
        postMessage(myVar);
      };
      // Rest of your worker code goes here.
    </script>
    <script>
      // This script WILL be parsed by JS engines because its MIME type is text/javascript.

      // In the past blob builder existed, but now we use Blob
      const blob = new Blob(Array.prototype.map.call(
        document.querySelectorAll("script[type='text\/js-worker']"),

>         (script) => { return oScript.textContent; }), 

Reviewer: ```suggestion
        (script) => script.textContent, 
```

    </style>
  </head>
  <body>
    <form>
      <div>
        <label for="number">Enter a number that is an index position in the fibonacci sequence to see what number is in that position (e.g. enter 5 and you'll get a result of 8 — fibonacci index position 5 is 8).</label>
        <input type="number" id="number">
      </div>
      <div>
        <input type="submit">
      </div>
    </form>

    <p id="result"></p>

    <script>
      const form = document.querySelector('form');
      const input = document.querySelector('input[type="number"]');
      const result = document.querySelector('p#result');
      const worker = new Worker('fibonacci.js');

      worker.onmessage = (event) => {
        result.textContent = event.data;
        console.log(`Got: ${event.data}`);
      };

      worker.onerror = (error) => {

>         console.log(`Worker error: ${error.message`);

Reviewer: ```suggestion
        console.log(`Worker error: ${error.message}`);
```
